omim2obo/main.py

```diff
@@ -345,6 +346,15 @@ def omim2obo(use_cache: bool = False):
             if p_mim_type == 'GENE':  # *
                 print(mim_type_err, file=sys.stderr)  # OMIM recognized as data quality issue. Fixed 2024/11. Failsafe.
 
+            # Self-referential special cases.
+            # Known cases: https://docs.google.com/spreadsheets/d/1hKSp2dyKye6y_20NK2HwLsaKNzWfGCMJMP52lKrkHtU/
+            if has_self_ref_assocs(p_mim, gene_phenotypes):
+                if p_mim not in ('159595', '182280', '607107', '615830'):  # previously known cases
```

Comment:
@joeflack4 I know these are the previously known cases (and perhaps they will always be "self-referential", but since you have the method `has_self_ref_assocs`, why are you explicitly hard-coding these entries/OMIM identifiers and not letting them be excluded based on that the method identifies these as "self-referential"?

Reply:
They are being excluded based on the logic, not the hard-coding:

```python
if has_self_ref_assocs(p_mim, gene_phenotypes):
    if p_mim not in ('159595', '182280', '607107', '615830'):  # previously known cases
        # todo: Add to review.tsv?
        print(f'Unexpected new self-referential case: OMIM:{p_mim}: {p_lab} (Gene: OMIM:{gene_mim})',
              file=sys.stderr)
    continue
```

The only purpose of this is to alert is in case any new cases arise. Perhaps new cases might not be quite the same as old cases, and we might review them. So I have it print a warning. And I have a `todo` there wondering if I should add a `review.tsv` case for these. Let me know what you think.

We could also just not have the warning, but I feel more comfortable knowing if more such new cases arise.

Reply:
Are these all in the `review.tsv` file now?

Reply:
@twhetzel Yes. The code has been changed:
1. These are again not being filtered out.
2. They are being added as `review.tsv` cases.
3. Documentation for the case added in the `README.md`.